### DISCUSSION TITLE
Sveltekit store that makes fetch call

### DISCUSSION THREAD

**Discussion by JonathonRP** [sveltejs/kit#8622] [Q&A] [answered]
Where is the fetch call made from, client or server?

**Comment by teemingc** [comment, marked as answer]
Depends where the store's function is called.

Inside a `.svelte` file and during SSR, it would run on the server.

When inside a server load function, it runs on the server.

Otherwise, it would run on the client.

**Reply by 205g0** [reply]
> When inside a server load function, it runs on the server

I thought that stores in `load`, `+page.ts` and `+page.server.ts` shouldn't be used anyway.

**Reply by teemingc** [reply]
I think it's possible to create a request specific store inside a `load` function and return it as data to be used client-side.

Otherwise, you can intentionally have a global store / value when initialising it outside the `load` function on the server.
https://kit.svelte.dev/docs/load#shared-state

(no idea if this is right, but this is how I understand it)

**Reply by JonathonRP** [reply]
Yeah I tried the load function a few times and passing the store through data props needs to be JSON serializable. But maybe using event.locals might work better

**Reply by teemingc** [reply]
Ah right. I forgot to mention you can only return a store when using a shared load function such as `+page.js`. They don't work on the server load ones.

**Reply by JonathonRP** [reply]
> Ah right. I forgot to mention you can only return a store when using a shared load function such as `+page.js`. They don't work on the server load ones.

Can you share an example code? I could have sworn I used a `page.ts`.

**Reply by teemingc** [reply]
[Here's an example.](https://stackblitz.com/edit/sveltejs-kit-template-default-xkjfm3?file=src%2Froutes%2F%2Blayout.svelte,src%2Froutes%2F%2Bpage.svelte&terminal=dev)

I first read about passing stores through `load` in this article https://www.loopwerk.io/articles/2022/sveltekit-architecture/
The syntax is different in the article because it uses the old load functions API (but the idea is there!)
